analyses/cell-type-neuroblastoma-04/exploratory-notebooks/singler-results.Rmd

```diff
@@ -620,6 +631,153 @@ ggplot(compare_annotations_expr_df) +
 The left-most gene set on the x-axis is `Neuroendocrine`, and normal cell types follow.
 This plot suggests that the normal cells `SingleR` is picking up are likely to be normal cells: normal gene expression for non-NE cells is higher compared to NE cells, and we see the opposite trend for the `Neuroendocrine` marker genes.
 
+
+## Inferences from smaller vs. larger libraries 
+
+In this section, we briefly look at annotation results for libraries with fewer than 1000 cells which may be lower quality than other libraries.
+
+Those libraries are:
+
+```{r}
+libraries_low_cells <- celltype_recoded_df |>
+  dplyr::filter(annotation_type == "singler") |>
+  dplyr::select(-annotation_type) |>
+  dplyr::count(library_id, name = "total_cells") |>
+  dplyr::filter(total_cells < 1000) 
+libraries_low_cells
+```
+
+### Distirbution of cell types
+
+
+```{r, fig.width = 10, fig.height = 4}
+celltype_recoded_wide_df |>
+  dplyr::mutate(
+    library_size = ifelse(library_id %in% libraries_low_cells$library_id, "<1000 cells", ">1000 cells")
+  ) |>
+  ggplot() + 
+  aes(x = library_id, fill = singler) + 
+  geom_bar(position = "fill") + 
+  scale_fill_manual(values = recoded_celltype_pal) + 
+  facet_wrap(vars(library_size), scales = "free_x") + 
+  scale_y_continuous(expand = c(0,0)) +
+  theme(
+    legend.position = "bottom",
+    axis.text.x = element_blank()
+  )
+```
+Although there are only four libraries with <1000 cells so it is challenging to make definitive conclusions, it does not seem like the overall cell type distributions are very different between smaller and larger libraries.
+
+### Heatmap
+
+This section shows a heatmap comparing annotations from these libraries only to the consensus annotations.
+
+```{r, fig.width = 8, fig.height = 8}
+make_jaccard_heatmap(
+  celltype_recoded_wide_df |> dplyr::filter(library_id %in% libraries_low_cells$library_id),
```

Comment:
```suggestion
low_cells_df <- celltype_recoded_wide_df |> 
  dplyr::filter(library_id %in% libraries_low_cells$library_id)

make_jaccard_heatmap(
  low_cells_df,
```
This is so minor but I don't love filtering on one line as input to the function. We should do the filtering first and then provide it as input.